user-interface/src/lib/components/LoadingSpinner.tsx

```diff
@@ -4,14 +4,25 @@ interface LoadingSpinnerProps {
   id?: string;
   caption?: string;
   className?: string;
+  hidden?: boolean;
+  height?: string;
 }
 
 export function LoadingSpinner(props: LoadingSpinnerProps) {
-  const id = props.id || 'loading-spinner';
+  const id = props.id || 'loading-spinner-' + Date.now();
+
   return (
-    <div id={id} className={`loading-spinner ${props.className ?? ''}`} data-testid={id}>
+    <div
+      id={id}
+      className={`loading-spinner ${props.className ?? ''}`}
+      data-testid={id}
+      style={{
+        visibility: props.hidden === true ? 'hidden' : 'visible',
+        height: props.height ? props.height : undefined,
+      }}
+    >
       <svg
-        id="loading-spinner-svg"
+        id={`${id}-svg`}
```

Comment:
```suggestion
        id={`loading-spinner-${id}-svg`}
```

Reply:
Thought it might be useful to denote what the svg is. Unless its being passed thru from the id.